data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -90,6 +95,7 @@ public PipelineModel(
         this.source = source;
         this.buffer = buffer;
         this.processors = processors;
+        this.router = router != null ? router : new ArrayList<>();
```

Comment:
I'd probably want to make `processors` return an empty list too.

Right now, this class has two constructors. One used by Jackson and the other used by other code in Data Prepper. I'd like to replace the second constructor with a builder pattern. It would be too much for this PR though. In that scenario, we could always use empty arrays instead of null to make development easier.

Reply:
I created #1669 to track these two changes. I propose there that `processors` should return an empty list.